examples/gnrc_tcp_cli/main.c

```diff
@@ -0,0 +1,235 @@
+        switch (ret) {
+            case 0:
+                DEBUG("TID=%"PRIi32" : gnrc_tcp_open_active() : 0 : ok\n", tid);
+                break;
+
+            case -EISCONN:
+                printf("TID=%"PRIi32" : gnrc_tcp_open_active() : -EISCONN\n", tid);
+                continue;
+
+            case -EINVAL:
+                printf("TID=%"PRIi32" : gnrc_tcp_open_active() : -EINVAL\n", tid);
+                continue;
+
+            case -EAFNOSUPPORT:
+                printf("TID=%"PRIi32" : gnrc_tcp_open_active() : -EAFNOSUPPORT\n", tid);
+                continue;
+
+            case -EADDRINUSE:
+                printf("TID=%"PRIi32" : gnrc_tcp_open_active() : -EADDRINUSE\n", tid);
+                continue;
+
+            case -ECONNREFUSED:
+                printf("TID=%"PRIi32" : gnrc_tcp_open_active() : -ECONNREFUSED : retry 5sec\n", tid);
+                xtimer_sleep(5);
+                continue;
+
+            case -ENOMEM:
+                printf("TID=%"PRIi32" : gnrc_tcp_open_active() : -ENOMEM\n", tid);
+                continue;
+
+            case -ETIMEDOUT:
+                printf("TID=%"PRIi32" : gnrc_tcp_open_active() : -ETIMEDOUT\n", tid);
+                continue;
+
+            default:
+                printf("TID=%"PRIi32" : gnrc_tcp_open_active() : %"PRIi32"\n", tid, ret);
+                continue;
+        }
+
+        /* Fill Buffer with a test pattern */
+        for (uint32_t i=0; i < sizeof(bufs[tid]) - 1; i+= 2){
+            bufs[tid][i] = 0xFF;
+            bufs[tid][i+1] = 0x00;
+        }
+
+        /* Send Data, stop if errors were found */
+        for (sent = 0; sent < sizeof(bufs[tid]) && ret >= 0; sent += ret) {
+            ret = gnrc_tcp_send(&tcb, bufs[tid] + sent, sizeof(bufs[tid]) - sent, 0);
+            switch (ret) {
+                case -ENOTCONN:
+                    printf("TID=%"PRIi32" : gnrc_tcp_send() : -ENOTCONN\n", tid);
+                    break;
+
+                case -ECONNABORTED:
+                    printf("TID=%"PRIi32" : gnrc_tcp_send() : -ECONNABORTED\n", tid);
+                    break;
+
+                case -ETIMEDOUT:
+                    printf("TID=%"PRIi32" : gnrc_tcp_send() : -ETIMEDOUT\n", tid);
+                    break;
+
+                case -ECONNRESET:
+                    printf("TID=%"PRIi32" : gnrc_tcp_send() : -ECONNRESET\n", tid);
+                    break;
+
+                default:
+                    if (ret >= 0) {
+                        DEBUG("TID=%"PRIi32" : gnrc_tcp_send() : %"PRIi32" Bytes sent\n", tid, ret);
+                    }
+                    else {
+                        printf("TID=%"PRIi32" : gnrc_tcp_send() : %"PRIi32"\n", tid, ret);
+                        break;
+                    }
+              }
+        }
+
+        /* Receive Data, stop if errors were found */
+        for (rcvd = 0; rcvd < sizeof(bufs[tid]) && ret >= 0; rcvd += ret) {
```

Comment:
The next one i am pushing ;). I want to wait until I have a build result (too see if the current state is okay)